android/modules/media/src/java/ti/modules/titanium/media/TiCameraActivity.java

```diff
@@ -495,24 +496,33 @@ private static File writeToFile(byte[] data, boolean saveToGallery) throws Throw
 
 	static public void takePicture()
 	{
-		String focusMode = camera.getParameters().getFocusMode();
-		if (!(focusMode.equals(Parameters.FOCUS_MODE_EDOF) || focusMode.equals(Parameters.FOCUS_MODE_FIXED) || focusMode
-			.equals(Parameters.FOCUS_MODE_INFINITY))) {
-			AutoFocusCallback focusCallback = new AutoFocusCallback()
-			{
-				public void onAutoFocus(boolean success, Camera camera)
-				{
-					camera.takePicture(shutterCallback, null, jpegCallback);
-					if (!success) {
-						Log.w(TAG, "Unable to focus.");
-					}
-					camera.cancelAutoFocus();
-				}
-			};
-			camera.autoFocus(focusCallback);
-		} else {
-			camera.takePicture(shutterCallback, null, jpegCallback);
-		}
+	    try {
+	        String focusMode = camera.getParameters().getFocusMode();
+	        if (!(focusMode.equals(Parameters.FOCUS_MODE_EDOF)
+	                || focusMode.equals(Parameters.FOCUS_MODE_FIXED) || focusMode
+	                .equals(Parameters.FOCUS_MODE_INFINITY))) {
+	            AutoFocusCallback focusCallback = new AutoFocusCallback()
+	            {
+	                public void onAutoFocus(boolean success, Camera camera)
+	                {
+	                    camera.takePicture(shutterCallback, null, jpegCallback);
+	                    if (!success) {
+	                        Log.w(TAG, "Unable to focus.");
+	                    }
+	                    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
```

Comment:
I think we should leave a comment here saying "cancelAutoFocus" causes the camera to crash on M (probably due to discontinued support of android.hardware.camera) and we should move to android.hardware.camera2.